Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndo.cpp

```diff
@@ -33,12 +33,23 @@ namespace AzToolsFramework
         void PrefabUndoInstance::Capture(
             const PrefabDom& initialState,
             const PrefabDom& endState,
-            TemplateId templateId)
+            InstanceOptionalReference instanceRef)
```

Comment:
Rather than add this logic directly to the capture logic, I suggest you move this to PrefabUndoHelpers::UpdatePrefabInstance(). That way, if we want to create an undo node using an already updated instance, they can call the PrefabUndoHelpers version. If we want to only operate with dom and not update any instance, we can continue to call PrefabUndoInstance::Capture(). The later case will be useful for workflows like duplicate entities, which is dom authoritative.

Reply:
That's a good suggestion!